### PR TITLE
[PATCH v4] api: remove deprecated old APIs

### PR DESCRIPTION
**Pull request by MatiasElo**
Remove old APIs which have been deprecated before 2020. Also, properly deprecate scheduling priority level defines `ODP_SCHED_PRIO_HIGHEST`, `ODP_SCHED_PRIO_NORMAL`, `ODP_SCHED_PRIO_LOWEST`, and `ODP_SCHED_PRIO_DEFAULT`, which were marked as to be deprecated in 2018.

V3:
- Removed deprecated odp.h header

### CODE REVIEW

**Review by Malvika-Gupta — COMMENTED** [review]
You can remove the variable aes_gcm as well. This was introduced for supporting deprecated APIs

File: platform/linux-generic/odp_crypto_openssl.c [hunk 1]
```diff
@@ -2419,14 +2379,6 @@ odp_crypto_session_create(const odp_crypto_session_param_t *param,
 	case ODP_AUTH_ALG_AES_XCBC_MAC:
 		rc = process_aesxcbc_param(session, EVP_aes_128_ecb());
 		break;
-#if ODP_DEPRECATED_API
-	case ODP_AUTH_ALG_AES128_GCM:
-		if (param->cipher_alg == ODP_CIPHER_ALG_AES128_GCM)
-			aes_gcm = 1;
```
Comment (Malvika-Gupta) [inline platform/linux-generic/odp_crypto_openssl.c:2425]: You can actually remove this variable, aes_gcm. The way I understand, it was introduced to support deprecated ODP_AUTH_ALG_AES128_GCM when the switch case falls through

**Review by JereLeppanen — APPROVED** [review]

File: platform/linux-generic/odp_crypto_openssl.c (hunk 1 above)
Comment (MatiasElo) [reply platform/linux-generic/odp_crypto_openssl.c:2425]: Thanks, fixed in V2.

**Comment by MatiasElo** [comment]
>  I saw in the commits that the deprecated APIs have been removed in `odp_crypto_openssl.c` but not in the aarch64 specific file, `platform/linux-generic/arch/aarch64/odp_crypto_armv8.c`. Can you please include those changes too? If you'd like, I can include it as a commit in the PR #1354 . Thanks!

Hi, I'd implemented this PR before the Arm crypto was merged and I missed it while rebasing. Fixed in V2.

File: include/odp/api/spec/crypto.h [hunk 2]
```diff
@@ -592,9 +592,6 @@ typedef struct odp_crypto_session_param_t {
 	 */
 	union {
 #if ODP_DEPRECATED_API
-		/** @deprecated Use cipher_iv */
-		odp_crypto_iv_t ODP_DEPRECATE(iv);
```
Comment (psavol) [inline include/odp/api/spec/crypto.h:633]: Commit log text refers to wrong field: "Remove deprecated 'cipher_iv' field...". Should be "iv" field.

**Review by psavol — APPROVED** [review]
one minor comment on a commit log entry

File: include/odp/api/spec/crypto.h (hunk 2 above)
Comment (MatiasElo) [reply include/odp/api/spec/crypto.h:633]: Fixed in V3.

**Pull request merged**